<launch>
    <arg name="ROBOT_NAME" default="turtlebot3"/>
    <!--megarover turtlebot3 beego-->
    <arg name="IS_SIMULATOR" default="true"/>
    <arg name="USE_RVIZ" default="false"/>
    <arg name="OUT_PUT" default="screen"/>
    <arg name="FRAME_ID/GLOBAL" default="map"/>
    <arg name="FRAME_ID/ROBOT_BASE" default="potbot"/>
    <arg name="FRAME_ID/LIDAR" default="lidar"/>
    <arg name="TOPIC/SCAN" default="scan"/>
    <arg name="TOPIC/ODOM" default="position"/>

    <node pkg="potbot_localization" name="potbot_localization_lo" type="potbot_localization_lo" output="screen">
        <param name="ROBOT_NAME" value="$(arg ROBOT_NAME)"/>
        <param name="LOCALIZATION_METHOD" value="dead_reckoning"/>
        <!--dead_reckoning particle_filter-->
        <param name="IS_SIMULATOR" value="$(arg IS_SIMULATOR)"/>
        <param name="USE_RVIZ" value="$(arg USE_RVIZ)"/>
        <param name="COVARIANCE_VV" value="0.05"/>
        <param name="COVARIANCE_VOMEGA" value="0.025"/>
        <param name="COVARIANCE_OMEGAOMEGA" value="0.05"/>
        <!-- <param name="INITIAL_POSE_X" value="-9.5"/>
        <param name="INITIAL_POSE_Y" value="-0.6"/>
        <param name="INITIAL_POSE_THETA" value="-1.5708"/> -->
        <param name="INITIAL_POSE_X" value="0"/>
        <param name="INITIAL_POSE_Y" value="0"/>
        <param name="INITIAL_POSE_THETA" value="0"/>
        <param name="FRAME_ID/GLOBAL" value="$(arg FRAME_ID/GLOBAL)"/>
        <param name="FRAME_ID/ROBOT_BASE" value="$(arg FRAME_ID/ROBOT_BASE)"/>
        <param name="FRAME_ID/LIDAR" value="$(arg FRAME_ID/LIDAR)"/>
        <param name="TOPIC/SCAN" value="$(arg TOPIC/SCAN)"/>
        <param name="TOPIC/ODOM" value="$(arg TOPIC/ODOM)"/>
    </node>
</launch>
